src/backend/distributed/commands/table.c

```diff
@@ -1396,14 +1410,23 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 		}
 		else if (alterTableType == AT_AddColumn)
 		{
-			/*
-			 * TODO: This code path is nothing beneficial since we do not
-			 * support ALTER TABLE %s ADD COLUMN %s [constraint] for foreign keys.
-			 * However, the code is kept in case we fix the constraint
-			 * creation without a name and allow foreign key creation with the mentioned
-			 * command.
-			 */
 			ColumnDef *columnDefinition = (ColumnDef *) command->def;
+
+			HeapTuple columnTuple = SearchSysCacheAttName(leftRelationId,
+														  columnDefinition->colname);
+			bool columnExists = HeapTupleIsValid(columnTuple);
+			if (columnExists)
+			{
+				/*
+				 * Return NIL regardless of whether IF NOT EXISTS is provided.
+				 * If the column exists and IF NOT EXISTS is provided, standard
+				 * process utility will do nothing. And if that option was not
+				 * provided, standard process utility will throw an error anyway.
+				 */
+				ReleaseSysCache(columnTuple);
+				return NIL;
```

Comment:
shouldn't we continue looping over other subcommands?

they might still need to be applied in case of IF NOT EXISTS

Reply:
Or maybe we should return NIL if it's the only subcommand and the column already exists. And if it's not the only subcommand and the column already exists, then we should continue the loop without setting deparseAT to true.

Reply:
Or maybe for simplicity, we shouldn't return NIL here but teach deparser to add IF NOT EXISTS when needed.

Reply:
> Or maybe for simplicity, we shouldn't return NIL here but teach deparser to add IF NOT EXISTS when needed.

seems like a good idea in general